lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2460,11 +2459,11 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
   } else {
     // 'wrt:' is specified. Validate and collect the selected parameters.
     int lastIndex = -1;
-    for (size_t i = 0; i < uncheckedWrtParams.size(); i++) {
-      auto paramLoc = uncheckedWrtParams[i].getLoc();
-      switch (uncheckedWrtParams[i].getKind()) {
-      case AutoDiffParameter::Kind::Index: {
-        unsigned index = uncheckedWrtParams[i].getIndex();
+    for (size_t i = 0; i < parsedWrtParams.size(); i++) {
```

Comment:
Some improvement:
```suggestion
    for (unsigned i : indices(parsedWrtParams)) {
```

Reply:
Thanks! Done in https://github.com/apple/swift/pull/21837/commits/6fe2c6458258a66d31c72730d89aed36f297996d.